primap2/_merge.py

```diff
@@ -0,0 +1,331 @@
+from typing import Optional
+
+import xarray as xr
+from loguru import logger
+
+from ._accessor_base import BaseDataArrayAccessor, BaseDatasetAccessor
+
+
+# general functions
+def merge_with_tolerance_core(
+    da_start: xr.DataArray,
+    da_merge: xr.DataArray,
+    tolerance: Optional[float] = 0.01,
+    error_on_discrepancy: Optional[bool] = True,
+) -> xr.DataArray:
+    """
+    Merge two DataArrays with a given tolerance for descrepancies in values
+    present in both DataArrays. If values from the data to merge are already
+    present in da_start they are treated as equal if the relative difference
+    is below the tolerance threshold. The result will use the values present
+     in da_start.
+
+    The function determines non-unique coordinates and iterates over them to
+    try merging for individual values of the coordinates. Where merge for a value
+    fails because of unequal data, the function recursively dives into the other
+    coordinates until single time-series are left which are then analyzed for each
+    time step and errors are logged as desired.
+
+    The function assumes dataarrays have been checked for identical coordinates
+    etc as it is called by DataArray.pr.merge() and Dataset.pr.merge() (and itself).
+
+
+    Parameters
+    ----------
+    da_start: xr.DataArray
+        data to merge into
+    da_merge: xr.DataArray
+        data to merge into ds_start
+    tolerance: float (optional), default = 0.01
+        The tolerance to use when comparing data
```

Comment:
I think the docs everywhere must state that this is a *relative* tolerance, so the default value is 1%. Right?
Also, it should probably be specified that this is relative to values in `da_start`.

Both is already in the general function docs, but should be added to the Parameter docs as people are most likely to look there if they have questions about the tolerance.

Reply:
right. Added that information